### PR TITLE
Implement BuyingReservedCardsController

### LINKED ISSUE #79: Implement `BuyingReservedCardsController`
Our goal is to allow users to purchase any reserved card in their hand.

To achieve this, we need to implement the `BuyingReservedCardsController` class, similar to the `BuyingRevealedCardsController`. This class contains a public method called `buyReservedCard`, which serves as the entry point for this feature. When a user wants to buy a card, they will provide the UUID of the desired card. The method will then construct a message and send it to the server.

The server will respond with a message containing information about the event, including the new set of player tokens and the updated player's reserved cards. It is important to collect all of this information and update the game model accordingly.

Please note that we will not be updating the view or objects on the screen at this time, as the view is not yet complete. This can be done separately.

Here's the currently empty class:
https://github.com/Splendor-Mobile-Game/ClientSide/blob/c66cfbfdb7babb6582a749b1289b7b1449119650/app/src/main/java/com/example/splendormobilegame/activities/GameActivity/BuyingReservedCardsController.java#L13-L73

### PR DESCRIPTION
**Pull request by EaZZyHK**
Solved #79
I've used a method that is not yet implemented and waits for approval in another pull request (#87). I could not test it tho.

### CODE REVIEW

File: app/src/main/java/com/example/splendormobilegame/activities/GameActivity/BuyingReservedCardsController.java [hunk 1]
```diff
@@ -44,15 +61,39 @@ public class BuyReservedCardMessageHandler extends UserReaction {
...
+            // Update the model
+            Room room = Model.getRoom();
+            Game game = room.getGame();
+            User buyer = room.getUserByUuid(responseData.buyer.userUuid);
+            Card boughtCard = game.getCardByUuid(responseData.buyer.cardUuid);
+            ReservedCard tempReservedCard = new ReservedCard(boughtCard, buyer, false);
+
+
+            buyer.addCard(boughtCard);
+            game.removeReservedCard(buyer, tempReservedCard);
```
Comment (kestivvi) [inline app/src/main/java/com/example/splendormobilegame/activities/GameActivity/BuyingReservedCardsController.java]: You may want to create new method in the `Game` like `removeReservedCard(UUID userUuid, UUID cardUuid)`. This method will search the reservedCards for a match and remove it then.

**Review by kestivvi — CHANGES_REQUESTED** [review]
Nice 🎂

File: app/src/main/java/com/example/splendormobilegame/activities/GameActivity/BuyingReservedCardsController.java (hunk 1 above)
Comment (EaZZyHK) [reply app/src/main/java/com/example/splendormobilegame/activities/GameActivity/BuyingReservedCardsController.java]: I've made it your way 👍

**Review by kestivvi — APPROVED** [review]
Your promptness is astonishing 🎉🎉

**Pull request merged**